### PR TITLE
revdep_check verifies integrity of already downloaded package files

### PR DESCRIPTION
**Pull request by krlmlr**
to avoid surprises

### CODE REVIEW

**Comment by hadley** [comment]
Can you please merge/rebase?

**Comment by krlmlr** [comment]
Rebased.

File: R/decompress.r [hunk 1]
```diff
@@ -17,17 +17,28 @@ source_pkg <- function(path, subdir = NULL, before_install = NULL) {
     stop("Does not appear to be an R package (no DESCRIPTION)", call. = FALSE)
   }
 
+  list(pkg_path = pkg_path, bundle = bundle)
+}
+
+is_source_pkg <- function(path, subdir = NULL) {
+  res <- try(source_pkg_info(path = path, subdir = subdir), silent = TRUE)
```
Comment (hadley) [inline R/decompress.r]: Could you eliminate use of `try()` here please?

Comment (hadley) [inline R/decompress.r]: Think you missed this comment

**Comment by krlmlr** [comment]
Rebased, now using tryCatch() instead of try().

File: R/decompress.r [hunk 2]
```diff
@@ -17,17 +17,30 @@ source_pkg <- function(path, subdir = NULL, before_install = NULL) {
...
   }
 
+  list(pkg_path = pkg_path, bundle = bundle)
+}
+
+is_source_pkg <- function(path, subdir = NULL) {
+  tryCatch({
+    source_pkg_info(path = path, subdir = subdir)
+    TRUE
+  }, error = function(e) return(FALSE))
```
Comment (hadley) [inline R/decompress.r]: Don't need explicit return here, but it's not a blocker.

**Comment by hadley** [comment]
Thanks!

**Pull request merged**